### PR TITLE
Fix protoc java stub generation

### PR DESCRIPTION
**Pull request by szab100**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -86,6 +77,35 @@
...
+          <artifactId>lifecycle-mapping</artifactId>
+          <version>1.0.0</version>
+          <configuration>
+            <lifecycleMappingMetadata>
+              <pluginExecutions>
+                <pluginExecution>
+                  <pluginExecutionFilter>
+                    <groupId>kr.motd.maven</groupId>
+                    <artifactId>os-maven-plugin</artifactId>
+                    <versionRange>[1.5.0.Final,)</versionRange>
```
Comment (dturbay) [inline pom.xml]: I believe fixed version is the best option. With concrete version for all deps/plugins you can have reproducible builds.

**Comment by szab100** [comment]
This CL fixes the following:

 - the .proto files were located into java source folders, moved them into src/main/proto as per specification of maven-protoc plugin
 - the os-maven-plugin was accidentally added twice to pom.xml, only one is required
 - Eclipse was not able to run the protoc stub generation, because the os-maven-plugin maven extension was not executed by m2e (eclipse maven integration). Fortunately, the os-maven-plugin can be used as a maven plugin (rather than an extension). The plugin's 'detect' goal is now hooked into the earliest 'initialize' maven phase. In order to allow m2e to execute this execution hook, we need to add this m2e pom.xml snippet. This way eclipse runs os-maven-plugin on build, therefore resolving the os-architecture property and the protoc compiler plugin can generate the java stub sources.

**Comment by dturbay** [comment]
Ok, I understand the issue now but for me this cl is still controversial.You added kind of hack for an eclipse (1 of many IDEs).
 - Have you tried this plugin in Intellij Idea? May be we have to add some hacks for idea too.
 - Is it possible to build project outside of eclipse and still work in eclipse?
 - Is it possible to configure eclipse to disable that hook security restriction?

**Comment by szab100** [comment]
> Ok, I understand the issue now but for me this cl is still controversial.You added kind of hack for an eclipse (1 of many IDEs).

Well, like it or not, Eclipse still seems to be (by far) the top (java) IDE: https://trends.google.com/trends/explore?geo=US&q=%2Fm%2F01fs1d,%2Fm%2F03v0mn,%2Fm%2F01fchg

And I wouldn't say it's a hack, it's an extra configuration for Eclipse. Other than adding a few more jars downloaded by Maven, it has no effect on a console maven build or using with other IDEs.

> * Have you tried this plugin in Intellij Idea? May be we have to add some hacks for idea too.
Idea's Maven integration has less features than Eclipse's, it is basically just calling out to the command line maven utility. According to my tests, it works with the os-detect plugin out of the box, if the developer runs a maven build explicitly from Idea.

> * Is it possible to build project outside of eclipse and still work in eclipse?
Yes, it works, this pom entry has no effect on non-eclipse scenarios.

> * Is it possible to configure eclipse to disable that hook security restriction?
I don't know, It might be possible, but could be very complicated (m2e maven integration can be configured under .eclipse/plugins/..../../../ configuration files..). With this extra configuration in maven, allowing Eclipse to run this hook my intention is to make it simple and easy for Eclipse users to import and work with this project. In other words, to avoid complicated custom eclipse configurations and detailed instructions.

**Comment by szab100** [comment]
Update: There is a good chance that the m2e lifecycle config will be added to the 'os-maven-detect' plugin itself (inti its pom.xml), so the users won't need to do it. But it's unsure if it works that way.. See my conversation on the plugin's GH repo here: https://github.com/xolstice/protobuf-maven-plugin/issues/10#issuecomment-424532884

**Pull request merged**